pandas/core/arrays/categorical.py

```diff
@@ -1275,10 +1275,12 @@ def shift(self, periods):
             raise NotImplementedError("Categorical with ndim > 1.")
         if np.prod(codes.shape) and (periods != 0):
             codes = np.roll(codes, ensure_platform_int(periods), axis=0)
+            if fill_value is None:
+                fill_value = -1
```

Comment:
Hmm this isn't quite right... Codes are integers, but users don't deal with those directly, they deal with the values in the categories.

I think something like

```
elif key in self.categories:
    fill_value = self.categories.get_loc(key)
else:
    raise ValueError("'fill_value={}' is not present in this Categorical's categories".format(fill_value))
```

Reply:
sorry, not sure what `key` is in your example, can you specify?

Reply:
sorry, `fill_value`.

Reply:
added

Reply:
this is also not right, you can pass a null value here, which is de-facto missing,  so need a

```
if isna(fill_value):
    fill_value = -1
elif.....
```

Reply:
changed

Reply:
changed